csrc/executor_utils.cpp

```diff
@@ -1058,20 +1058,19 @@ class NvrtcCompileDriver {
     auto opts = getOptions();
     auto result = nvrtcCompileProgram(
         program, static_cast<int>(opts.size()), opts.data());
-
     size_t logsize = 0;
     NVFUSER_NVRTC_SAFE_CALL(nvrtcGetProgramLogSize(program, &logsize));
-    std::string log(logsize, 0);
+    std::string log;
```

Comment:
Please comment why this needs to be done in this way since the reason of the change is not explicitly clear.

I feel that using just a c-str would be less error prone for this case. Maybe something like:

```
std::vector<char> log_backing_buf(logsize);
char* log_buf = log_backing_buf.data();
NVFUSER_NVRTC_SAFE_CALL(nvrtcGetProgramLog(program, log_buf));
...
```

This way, there's no way to accidentally use the `std::string` object.

Reply:
updated.